src/scripts/main.js

```diff
@@ -1,11 +1,24 @@
 'use strict';
+const form = document.querySelector('form');
 
 function waitFor(element, eventName) {
-  // write your code here
+  if (element) {
+    return new Promise((resolve, reject) => {
+      element.addEventListener(eventName, () => {
+        const string = `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`;
+
+        resolve(string);
+      });
+    });
+  }
 }
+const messageDiv = document.createElement('div');
+
+messageDiv.classList.add('message');
 
 const printMessage = (message) => {
-  // write your code here
+  messageDiv.innerText = message;
+  form.append(messageDiv);
```

Comment:
The function 'printMessage' always appends a new message to the 'form'. This can lead to multiple messages being displayed at the same time. It would be better to clear the previous message before displaying a new one.